networkx/algorithms/dominating.py

```diff
@@ -42,34 +47,38 @@ def dominating_set(G, start_with=None):
     """
     all_nodes = set(G)
     if start_with is None:
-        v = set(G).pop() # pick a node
-    else:
-        if start_with not in G:
-            raise nx.NetworkXError('node %s not in G' % start_with)
-        v = start_with
-    D = set([v])
-    ND = set(G[v])
-    other = all_nodes - ND - D
-    while other:
-        w = other.pop()
-        D.add(w)
-        ND.update([nbr for nbr in G[w] if nbr not in D])
-        other = all_nodes - ND - D
-    return D
+        start_with = arbitrary_element(all_nodes)
+    if start_with not in G:
+        raise nx.NetworkXError('node %s not in G' % start_with)
```

Comment:
Can you convert this to use format(). Or make it `'node %s not in G' % (start_with,)`. Otherwise, if `start_with` is a 2-tuple (for example), this will fail.